warehouse/rate_limiting/__init__.py

```diff
@@ -162,7 +163,82 @@ def __eq__(self, other):
         )
 
 
+def ratelimit_view_deriver(view, info):
+    """
+    A general-purpose rate limit view deriver based on client's `remote_addr`.
+    https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/hooks.html#custom-view-derivers
+
+    Add `ratelimiter="<limit-string>"` to a `@view_config()`. Example:
+
+    ```
+    @view_config(
+        ...,
+        ratelimit="10/second",
+        ratelimit_by="user.id",  # Optional, defaults to "remote_addr"
+    )
+    ```
+    https://limits.readthedocs.io/en/stable/quickstart.html#rate-limit-string-notation
+    """
+    limit_str = info.options.get("ratelimit")
+    limit_by = info.options.get("ratelimit_by", "remote_addr")
+    if limit_str:
+
+        def wrapper_view(context, request):
+            limit_attr = getattr(request, limit_by)
+
+            ratelimiter = request.find_service(
+                IRateLimiter, name="rate_limiting.client"
+            )
+            # TODO: Is this Kosher?
+            # Override the default limits with the ones specified in the view
+            ratelimiter._limits = parse_many(limit_str)
```

Comment:
I don't know how I feel about this. I was trying to find a way to pass `limit_str` as a way to modify the Service's constructor call, but I can't figure out how to make it work with a pre-registered service factory other than doing it like this.